### ISSUE TITLE
Use ncvgeom in rendered mode

### ISSUE THREAD

**Issue by joseluis** [dankamongmen/notcurses#1844]
I noticed there's no tracking issue for this:

```c
// FIXME this ought be used in the rendered mode API as well; it's currently
// only used by direct mode. describes all geometries of an ncvisual--both those
// which are inherent, and those in a given rendering regime.
// ...
typedef struct ncvgeom
```

IIUC there are going to be 2 geometry outputs:
-  ncplane geometry:
    - total pixel geometry (h, w)
    - pixel geometry of an individual cell (h, w)
    - pixel geometry of the maximum representable bitmap (h, w)
-  ncvisual geometry:
    - pixel geometry of internal data (h, w)
    - terminal (or plane?) geometry when the ncvisual was created (rows, cols).
    - rendered pixel geometry (h, w)
    - rendered cell geometry (rows, cols)
    - pixel geometry of an individual cell (h, w) *{also in ncplane geom}*
    - pixel geometry of the maximum representable bitmap (h, w) *{also in ncplane geom}*
    - blitter that will be used

**Comment by dankamongmen** [comment]
i think this is basically #1684 ?

**Comment by joseluis** [comment]
> i think this is basically #1684 ?

Oh, maybe? It wasn't clear to me, sorry.

I also wanted to kind of confirm that was going to be the final API, since I was sort of figuring out the higher-level API... e.g.  for the plane geometry:

```
Geometry {
    x: 1920,
    y: 1034,
    cols: 192,
    rows: 47,
    bx: 1920,
    by: 1012,
    bcols: 192,
    brows: 46,
    cx: 10,
    cy: 22,
}
```

I'll wait for the visual geometry to be used on rendered mode to continue, you can close this issue if you believe it's duplicated.

**Comment by dankamongmen** [comment]
nah they're just related, it's not a dup

**Comment by dankamongmen** [comment]
this should be done before #1777

**Comment by dankamongmen** [comment]
OK, i'm going to go ahead and close this as a dup of #1684 after all, which I'm currently working on, and will be the One True Way Of Ascertaining Geometry as of abi3 (it'll be introduced before then, ASAP, probably 2.4.9).

**Comment by dankamongmen** [comment]
Duplicate of #1684.
